components/modal/aria-helper.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+let hiddenElements = [],
+	isHidden = false;
+
+/**
+ * Hides all elements in the body element from screen-readers except
+ * the provided element, script elements and elements that already have
+ * an `aria-hidden="true"` attribute.
+ *
+ * The reason we do this is because `aria-modal="true"` currently is bugged
+ * in Safari, and support is spotty in other browsers overall. In the future
+ * we should consider removing these helper functions in favor of
+ * `aria-modal="true"`.
+ *
+ * @param {Element} unhiddenElement The element that should not be hidden.
+ */
+export function hideApp( unhiddenElement ) {
+	if ( isHidden ) {
```

Comment:
Why is the condition (and the associated `isHidden` variable) needed? Are we expecting this function to be called excessively often so as to need the optimization shortcut? Would it break if we otherwise allow the function to proceed?

Reply:
It could be called multiple time in case there are multiple modals. Although this is not desired, various plugins could open modals making this check a worthwhile check in my opinion.